procedural/timeline/src/main/kotlin/gov/nasa/ammos/aerie/procedural/timeline/collections/ExternalEvents.kt

```diff
@@ -0,0 +1,31 @@
+package gov.nasa.ammos.aerie.procedural.timeline.collections
+
+import gov.nasa.ammos.aerie.procedural.timeline.Timeline
+import gov.nasa.ammos.aerie.procedural.timeline.BaseTimeline
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.Instance
+import gov.nasa.ammos.aerie.procedural.timeline.ops.*
+import gov.nasa.ammos.aerie.procedural.timeline.ops.coalesce.CoalesceNoOp
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.ExternalEvent
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.ExternalSource
+import gov.nasa.ammos.aerie.procedural.timeline.util.preprocessList
+
+/**
+ * A timeline of external events.
+ */
+data class ExternalEvents(private val timeline: Timeline<ExternalEvent, ExternalEvents>):
+    Timeline<ExternalEvent, ExternalEvents> by timeline,
+    NonZeroDurationOps<ExternalEvent, ExternalEvents>,
+    ParallelOps<ExternalEvent, ExternalEvents>
+{
+  constructor(vararg events: ExternalEvent): this(events.asList())
+  constructor(events: List<ExternalEvent>): this(BaseTimeline(::ExternalEvents, preprocessList(events, null)))
+
+  /** Filter by one or more types. */
+  fun filterByType(vararg types: String) = filter { it.type in types }
```

Comment:
This looks like yet another filtering API we're exposing to users. When would we expect users to use these functions v.s. querying while getting the `ExternalEvents` object?

Reply:
If there are conditional filters we want to apply on an `ExternalEvents` object, this is how we could do it without having to pull repeatedly from the `Plan` object.